fe/fe-core/src/main/java/com/starrocks/leader/LeaderImpl.java

```diff
@@ -590,6 +591,10 @@ private void finishPublishVersion(AgentTask task, TFinishTaskRequest request) {
             }
         }
         publishVersionTask.setIsFinished(true);
+        TransactionState txnState = publishVersionTask.getTxnState();
+        if (txnState != null) {
+            txnState.updatePublishTaskFinishTime();
```

Comment:
This is only valid if all tasks are finished normally and within Config.quorom_publish_wait_time_ms
Do we have a better way to make it more accurate?

Reply:
Yes, this is only to give a rough judge of the publish rpc cost of the txn, more accurate statistics like rpc cost of different be can be done, but i think currently it's not that necessary, so i will keep this logic unchanged for now.